### ISSUE TITLE
Pytorch Sweeps tracing - ttnn.add - Low PCC and other fails

### ISSUE THREAD

**Issue by KalaivaniMCW** [tenstorrent/tt-metal#14730]
Following parameters failed during Pytorch Sweeps tracing tests: for ttnn.add
(the below cases are ignoring the fails for Tensor Layout issues )
Test file: `tests/sweep_framework/sweeps/eltwise/binary/add/add_all_pytorch2.py`
Pytorch sweeps: [tracker](https://docs.google.com/spreadsheets/d/1zumGqgcrweLBY4wn1OzKCcf-DJDSiaj6_bDvjfT74JY/edit?usp=sharing)

<google-sheets-html-origin>
**Parameter** |   | **Error**
-- | -- | -- 
{'input_shape': {'self': [1, 71, 7, 7], 'other': [7, 7]}, |   | low pcc fail
{'input_shape': {'self': [920, 1, 256], 'other': [256]}, |   | low pcc fail
{'input_shape': {'self': [0, 1], 'other': [0, 1]}, |   | round_up: multiple must not be 0
{'input_shape': {'self': [0], 'other': [0]}, |   | round_up: multiple must not be 0
{'input_shape': {'self': [3234], 'other': [3234]}, |   | message list(expected_pytorch_result.shape)=[3234] vs list(actual_pytorch_result.shape)=[1, 3234]
{'input_shape': {'self': [8732], 'other': [8732]}, |   | message list(expected_pytorch_result.shape)=[8732] vs list(actual_pytorch_result.shape)=[1, 8732]
{'input_shape': {'self': [], 'other': []}, |   | incompatible function arguments. The following argument types are supported:
{'input_shape': {'self': [], 'other': 1}, |   | incompatible function arguments. The following argument types are supported:
**Below are fails that occur on L1 memory: 4 cases (both inputs on L1)**
{'input_shape': {'self': [13600, 1, 4], 'other': [1, 9, 4]}, |   | Out of Memory: Not enough space to allocate 27852800 B L1 buffer across 64 banks, where each bank needs to store 436224 B
{'input_shape': {'self': [1, 64, 480, 640], 'other': [1, 64, 480, 640]}, |   | Out of Memory: Not enough space to allocate 39321600 B L1 buffer across 64 banks, where each bank needs to store 614400 B
{'input_shape': {'self': [1, 256, 200, 272], 'other': [1, 256, 200, 272]}, |   | Out of Memory: Not enough space to allocate 33030144 B L1 buffer across 64 banks, where each bank needs to store 516096 B
{'input_shape': {'self': [1, 256, 180, 320], 'other': [1, 256, 180, 320]}, |   | Out of Memory: Not enough space to allocate 31457280 B L1 buffer across 64 banks, where each bank needs to store 491520 B

**Comment by ayerofieiev-tt** [comment]
@KalaivaniMCW , can you share some info on the PCC issue? Do we have any ideas on why this might happen?

**Comment by KalaivaniMCW** [comment]
@ayerofieiev-tt I debugged the two shapes that failed for PCC. It is because of the unequal ranks. the test passes if I unsqueeze the lesser-ranked shape before passing to the op.
I tried `ttnn::unsqueeze` to handle this in our cpp implementation but it has a restriction  _Currently supporing ROW-MAJOR tensors or TILE tensors with no padding_  and for eltwise binary we have the restriction that it supports TILE layout only. 

![Image](https://github.com/user-attachments/assets/e8af436d-0f76-4f66-9465-d82d99b1a833)

Shape 1
![Image](https://github.com/user-attachments/assets/3d95883a-5132-4d8a-a790-d896a4e1caf7)

Shape 2
![Image](https://github.com/user-attachments/assets/7f732918-37fc-4c1b-9c03-263b498e7615)

**Comment by eyonland** [comment]
@ntarafdar , is it possible for us to get ttnn::unsqueeze to work with padded shapes?

**Comment by ntarafdar** [comment]
@eyonland it should be working as it's a wrapper for reshape which handles it by removing padding

**Comment by KalaivaniMCW** [comment]
> [@KalaivaniMCW](https://github.com/KalaivaniMCW) can you make a pytest that illustrates the issue with unsqueeze so that we can troubleshoot this a bit better?

@eyonland Yes will do.

**Comment by KalaivaniMCW** [comment]
@ntarafdar @eyonland I am trying to unsqueeze the 2nd tensors in TILE layout

```
import pytest

import torch

import ttnn
from tests.ttnn.utils_for_testing import assert_with_pcc


@pytest.mark.parametrize(
    "shapes",
    [
        [[1, 71, 7, 7], [7, 7]],
        [[920, 1, 256],  [256]],
    ],
)
def test_unsqueeze_tile(device, shapes):
    torch.manual_seed(0)

    torch_input_tensor_a = torch.rand(shapes[0], dtype=torch.bfloat16)
    torch_input_tensor_b = torch.rand(shapes[1], dtype=torch.bfloat16)
    torch_output_tensor = torch.unsqueeze(torch_input_tensor_b, 0)
    print("torch_output_tensor", torch_output_tensor.shape)
    
    input_tensor_a = ttnn.from_torch(
        torch_input_tensor_a, layout=ttnn.TILE_LAYOUT, device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG
    )

    input_tensor_b = ttnn.from_torch(
        torch_input_tensor_b, layout=ttnn.TILE_LAYOUT, device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG
    )

    output_tensor = ttnn.unsqueeze(input_tensor_b, 0)
    print("output_tensor", output_tensor)
    output_tensor = ttnn.to_torch(output_tensor)

    assert ttnn.pearson_correlation_coefficient(torch_output_tensor, output_tensor) >= 0.99988
```
getting this error
![Image](https://github.com/user-attachments/assets/b4dbc7c4-4072-4c60-b36f-f8a070b9a552)
![Image](https://github.com/user-attachments/assets/481e50b6-41c3-4aa1-8a72-0ba79a6cf31c)

**Comment by KalaivaniMCW** [comment]
@ntarafdar @eyonland , instead of unsqueeze I am able to use `ttnn::reshape` to convert `ttnn.Shape([7[32], 7[32]])` to `Shape([1, 1, 7[32], 7[32]]) `

**Comment by KalaivaniMCW** [comment]
Merged #14803 
The two pcc fails mentioned here are now passing

**Comment by KalaivaniMCW** [comment]
Support for these Low PCC fail cases have been added in binary_ng https://github.com/tenstorrent/tt-metal/pull/15957 and will be available for the ttnn.add api when binary_ng is integrated with eltwise binary
